packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -11,9 +11,83 @@ import 'framework.dart';
 /// Return true to cancel the notification bubbling. Return false to allow the
 /// notification to continue to be dispatched to further ancestors.
 ///
+/// [NotificationListener] is useful when listening scroll events
+/// in [ListView],[NestedScrollView],[GridView] or any Scrolling widgets.
 /// Used by [NotificationListener.onNotification].
+
 typedef NotificationListenerCallback<T extends Notification> = bool Function(T notification);
 
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This example shows a [NotificationListener ] widget
+/// which its type is [ScrollNotification ], when there is scroll
```

Comment:
doc tests are failing because of typos here: https://github.com/flutter/flutter/pull/69524/checks?check_run_id=1339299491.

There shouldn't be a space when you do a doc reference in between your brackets `[]`